### DISCUSSION TITLE
FP8 training of LoRAs with torchao

### DISCUSSION THREAD

**Discussion by sayakpaul** [bghira/SimpleTuner#742] [General]
@bghira I was able to hack together a dirty script for FP8 LoRA training through `torchao`: https://gist.github.com/sayakpaul/2743810ed362cdc841b90956e2b155a1

But the results are not there yet. So, opening this discussion up for debugging this together. But some desired things already work:

* [Wrapping](https://gist.github.com/sayakpaul/2743810ed362cdc841b90956e2b155a1#file-faulty_sd3_dreambooth_lora_fp8-py-L1201) the `transformer` into an FP8-compatible module. 
* [Injecting](https://gist.github.com/sayakpaul/2743810ed362cdc841b90956e2b155a1#file-faulty_sd3_dreambooth_lora_fp8-py-L1215) LoRA config. 
* [`accelerator.prepare()`](https://gist.github.com/sayakpaul/2743810ed362cdc841b90956e2b155a1#file-faulty_sd3_dreambooth_lora_fp8-py-L1448) on the `transformer`. 

Training command and other necessary instructions are in the gist itself. 

Cc @AmericanPresidentJimmyCarter who might be interested too.

**Comment by bghira** [comment]
what about el honcho - the big cheese - loading a checkpoint?

**Reply by bghira** [reply]
can it resume training?

**Reply by sayakpaul** [reply]
Yeah it can. But it needs to train first :D

**Reply by bghira** [reply]
![image](https://github.com/user-attachments/assets/52247e37-4ef3-4ea1-b133-d1253dfe269e)

**Reply by sayakpaul** [reply]
I think I miscommunicated it. I mean it is training but we need to ensure the correctness of training results.
